archive.tf

```diff
@@ -10,15 +10,15 @@ data "external" "archive" {
   query = {
     build_command  = "${var.build_command}"
     build_paths    = "${jsonencode(var.build_paths)}"
-    module_relpath = "${local.module_relpath}"
+    module_relpath = "${path.module}"
     runtime        = "${var.runtime}"
-    source_path    = "${var.source_path}"
+    source_path    = "${path.cwd}/${var.source_path}"
```

Comment:
This was a weird one for me.

The python script was trying to cd into the module directory, and from there access the `hash.py` script (in the current directory), but also access the `source_path` script which comes as a relative path.

Now, I'm using this module through a second module of my own. And I set `source_path = "${module.path}/lambda.py"`.
Under terraform 0.12, this seems to produce something like "../../../../../../modules/ebs_recurrent_snapshot/lambda.py"

this is because the `ebs_recurrent_snapshot module (my own module) is 6 directories behind the actual code using it.

What also seems to happen is that python does `os.chdir` into `.terraform/ebs_recurrent_snapshot.lambda`, which is in fact a symlink to the real module location. But it follows the symlink by resolving it to the actual path.

So when python reaches the directory, the relative path to my `lambda.py` script no longer match

I have no idea if this was a changed introduced by terraform 0.12, by recent changes to this own repo, or by my own stupidity.

glad to take feedback here.
For now, providing a full path to the source file by using `path.cwd` seemed to work

Reply:
This is because they changed `path.module` from an absolute path to a relative path. I think I've fixed this in 6c1eb25440a8a24a622e068d27dff83911c41bab in by doing `os.path.abspath(query['source_path'])`.